scripts/run_commands.sh

```diff
@@ -0,0 +1,92 @@
+#
+# This scripts runs on the ec2 instance. It is uploaded to the EC2 instance by
+# test_commands.sh
+# The necessary env variables will be set by test_commands.sh using the
+# command line arguments provided to it. 
+#
+
+declare -i FAILURES
+FAILURES=0
+
+RED='\033[1;31m'
+GREEN='\033[1;32m'
+LRED='\033[0;31m'
+LGREEN='\033[0;32m'
+NC='\033[0m' # No Color
+
+expect_success() {
+	if "${@}" &>> ~/test_output.txt; then
+		echo -e "${GREEN}SUCCEEDED${NC}: ${LGREEN}${@}${NC}"
+	else
+		echo "-------"
+		echo -e "${RED}FAILED${NC}: ${LRED}${@}${NC}"
+		echo "-------"
+		FAILURES+=1
+	fi
+}
+
+# used for commands that whose arguments might contain sensitive information
+expect_success_no_log() {
+	if "${@}" &>> ~/test_output.txt; then
+		echo -e "${GREEN}SUCCEEDED${NC}: ${LGREEN}${1} ${2}${NC}"
+	else
+		echo "-------"
+		echo -e "${RED}FAILED${NC}: ${LRED}${1} ${2}${NC}"
+		echo "-------"
+		FAILURES+=1
+	fi
+}
+
+# install git and go
+echo "TYPE y|yes then enter to proceed."
+sudo yum install git go >> ~/test_log.txt
+# have to respond yes to prompt
+# get CLI
+export GOPATH="$HOME/go"
+go get github.com/aws/amazon-ecs-cli >> ~/test_log.txt
+cd go/src/github.com/aws/amazon-ecs-cli
+url="https://github.com/"
+url+=$gitname
+url+="/amazon-ecs-cli.git"
+git remote add fork $url &>> ~/test_log.txt
+git fetch fork &>> ~/test_log.txt
+git checkout "fork/${branch}"
+make build
```

Comment:
Is setting up git and go path really needed? Can you run make build first on your laptop and copy the binary to the EC2 instance?

You would run the following in order:
1. `make build` --> you will have a binary built already.
2. `make test` --> make sure unit tests works.
3. `make integration-test` --> copy the binary and the script to EC2 instance, and run all commands

Reply:
I considered this idea myself. It seems really really nice, but as I was implementing it I realized it will not work! See, the CLI supports both Mac and Linux- and the binaries are system dependent. So for example, if I use a Mac and then try to copy my built binary to an instance- it will not run on that instance, because it is built for the wrong platform.

Reply:
Ahh, I see. Can we use the binary from `make docker-build`? It will build two binaries, one for Mac and one for Linux.